src/v/model/transform.h

```diff
@@ -98,18 +154,31 @@ struct transform_offset_options
           envelope<latest_offset, serde::version<0>, serde::compat_version<0>> {
         bool operator==(const latest_offset&) const = default;
     };
-    // A transform can either start at the latest offset or at a timestamp.
+    // A transform can either start at the latest offset, at a timestamp, or at
+    // some delta from the start or end of an input partition.
     //
     // When a timestamp is used, a timequery is used to resolve the offset for
     // each partition.
-    serde::variant<latest_offset, model::timestamp> position;
+    //
+    // When transform_from_start is used, the offset_delta therein is added to
+    // the start offset for each partition.
+    //
+    // When transform_from_end is used, the offset_denta therein is subtracted
+    // from the latest offset of each partition.
+    serde::variant<
+      latest_offset,
+      model::timestamp,
+      model::transform_from_start,
+      model::transform_from_end>
+      position;
```

Comment:
We need a feature gate to prevent using the new position options on a partially rolled out cluster.

Reply:
Ah, right, because it's a new envelope.

Reply:
this is done. feels odd to have a whole feature flag for such a small change, but maybe we just don't do things like this very often.

Reply:
Maybe, or we just overlook it and customers are well behaved, but in this case if someone does this old brokers won't be able to read transform metadata - which would be real bad.